contexts/SummitData.js

```diff
@@ -1,112 +1,53 @@
 import {
   createContext,
   useContext,
-  useEffect,
   useMemo,
-  useState,
-  useReducer,
 } from "react";
-import {
-  currentWeather,
-  dailyWeather,
-  hourlyWeather,
-} from "@/lib/api/queries/weather";
 import PropTypes from "prop-types";
-import queryEfd from "@/lib/api/efd";
+import useEfd from "@/lib/api/efd";
 
 export const SummitDataContext = createContext({});
 
-const loadingReducer = (state, action) => {
-  const { type, dataSet } = action;
-
-  switch (type) {
-    case "complete":
-      return { ...state, [dataSet]: false };
-    case "loading":
-      return { ...state, [dataSet]: true };
-    default:
-      throw new Error(`Unhandled action type: ${type}`);
-  }
-};
-
-const defaultLoadState = {
-  currentData: true,
-  hourlyData: true,
-  dailyData: true,
-};
-
 export const SummitDataProvider = ({ children }) => {
-  const [currentData, setCurrentData] = useState();
-  const [hourlyData, setHourlyData] = useState();
-  const [dailyData, setDailyData] = useState();
-  const [error, setError] = useState(false);
-  const [loading, dispatch] = useReducer(loadingReducer, defaultLoadState);
-
-  const fetchCurrentWeather = () =>
-    new Promise((resolve, reject) => {
-      queryEfd(currentWeather)
-        .then((value) => {
-          setCurrentData(...value);
-          resolve(...value);
-        })
-        .catch((e) => {
-          reject(e);
-        })
-        .finally(() => {
-          dispatch({ type: "complete", dataSet: "currentData" });
-        });
-    });
-
-  const fetchHourlyWeather = () =>
-    new Promise((resolve, reject) => {
-      queryEfd(hourlyWeather)
-        .then((value) => {
-          setHourlyData(value);
-          resolve(value);
-        })
-        .catch((e) => {
-          reject(e);
-        })
-        .finally(() => {
-          dispatch({ type: "complete", dataSet: "hourlyData" });
-        });
-    });
-
-  const fetchDailyWeather = () =>
-    new Promise((resolve, reject) => {
-      queryEfd(dailyWeather)
-        .then((value) => {
-          setDailyData(value);
-          resolve(value);
-        })
-        .catch((e) => {
-          reject(e);
-        })
-        .finally(() => {
-          dispatch({ type: "complete", dataSet: "dailyData" });
-        });
-    });
-
-  useEffect(() => {
-    Promise.allSettled([
-      fetchCurrentWeather(),
-      fetchHourlyWeather(),
-      fetchDailyWeather(),
-    ]).catch((e) => {
-      console.error(e);
-      setError(true);
-    });
-  }, []);
+  const {
+    data: currentData,
+    isLoading: currentIsLoading,
+    isError: currentIsError,
+  } = useEfd("current");
+  const {
+    data: hourlyData,
+    isLoading: hourlyIsLoading,
+    isError: hourlyIsError,
+  } = useEfd("hourly");
+  const {
+    data: dailyData,
+    isLoading: dailyIsLoading,
+    isError: dailyIsError,
+  } = useEfd("daily");
```

Comment:
the `pages/api/efd` is now responsible for putting together the influxDB queries, based on the `type` that's specified via the internal API url that is fetched from `lib/api/efd`, so I think all of the promisey state handling fetching stuff can go away, and all happens elsewhere now, getting tucked behind this u`useEfd` hook.  Maybe?  It's honestly a bit hard to tell if I went about his last bit the correct way because the USDF is under scheduled maintenance so I can't yet fully test.